src/test/java/io/jenkins/plugins/analysis/warnings/IssuesRecorderITest.java

```diff
@@ -61,14 +63,151 @@ public void shouldCreateResultWithWarnings() {
     @Test
     public void shouldCreateUnstableResult() {
         FreeStyleProject project = createJobWithWorkspaceFile("eclipse.txt");
-        enableWarnings(project, publisher -> publisher.setUnstableTotalAll(7));
+        enableWarnings(project, publisher -> publisher.setUnstableTotalAll(7),
+                new ToolConfiguration("**/*issues.txt", new Eclipse()));
 
         AnalysisResult result = scheduleBuildAndAssertStatus(project, Result.UNSTABLE);
 
         assertThat(result).hasTotalSize(8);
         assertThat(result).hasOverallResult(Result.UNSTABLE);
     }
 
+    /**
+     * Sets the health threshold less then the unhealthy threshold and parse a file that contains exactly warnings. The
+     * healthReport should be null / empty because the healthReportDescriptor is not enabled with this setup
+     */
+    @Test
+    public void shouldCreateEmptyHealthReportForBoundaryMismatch() {
+        HealthReport report = createHealthReportTestSetupEclipse(5, 2);
+        assertThat(report).isNull();
+    }
+
+    /**
+     * Sets the health threshold equals to the unhealthy threshold and parse a file that contains exactly warnings. The
```

Comment:
Sets the value of the health threshold equal to the value of the unhealthy threshold and parses a file that contains exactly ? warnings